x-pack/solutions/observability/plugins/apm/server/routes/diagnostics/service_map/get_exit_spans_from_node.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { termQuery, rangeQuery, existsQuery, termsQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  SPAN_DESTINATION_SERVICE_RESOURCE,
+  SPAN_ID,
+  PARENT_ID,
+  SERVICE_NAME,
+} from '@kbn/apm-types';
+import type { APMEventClient } from '@kbn/apm-data-access-plugin/server';
+
+/**
+ * Fetch exit spans for a given service, grouped by destination resource, with a sample doc for each.
+ * @param start Start time (epoch ms)
+ * @param end End time (epoch ms)
+ * @param serviceName Service name to filter on
+ */
+export async function getExitSpansFromSourceNode({
+  apmEventClient,
+  start,
+  end,
+  sourceNode,
+  destinationNode,
+}: {
+  apmEventClient: APMEventClient;
+  start: number;
+  end: number;
+  sourceNode: { field: string; value: string };
+  destinationNode: { field: string; value: string };
+}) {
+  const response = await apmEventClient.search('diagnostics_get_exit_spans_from_node', {
+    apm: {
+      events: [ProcessorEvent.span],
+    },
+    track_total_hits: false,
+    size: 0,
+    query: {
+      bool: {
+        filter: [...rangeQuery(start, end), ...termQuery(sourceNode.field, sourceNode.value)],
+      },
+    },
+    aggs: {
+      matching_destination_resources: {
+        filter: {
+          term: {
+            [SPAN_DESTINATION_SERVICE_RESOURCE]: destinationNode.value,
```

Comment:
This won't work if the `destinationNode` is a service.

<img width="920" height="704" alt="image" src="https://github.com/user-attachments/assets/cc17f0d6-05e6-46ee-a4aa-e1353e07727a" />

`shipping` and `quote` have a connection, but the diagnostic says that they don't